lib/min_heap.rb

```diff
@@ -44,31 +52,61 @@ def to_s
   end
 
   # This method returns true if the heap is empty
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(1)
+  # Space complexity: O(1)
   def empty?
-    raise NotImplementedError, "Method not implemented yet..."
+    
+    if @store.nil? || @store.length == 0
+      return true
+    else 
+      return false
+    end
+
   end
 
   private
 
   # This helper method takes an index and
   #  moves it up the heap, if it is less than it's parent node.
   #  It could be **very** helpful for the add method.
-  # Time complexity: ?
-  # Space complexity: ?
+  # Time complexity: O(m) where m is height
+  # Space complexity: O(1), no persistent data structure is made beyond 1 temporary reassigned variable
   def heap_up(index)
```

Comment:
👍 , but I would give O(log n) for the time complexity because the tree is always balanced and the height is always log n.  The space complexity would be O(log n) due to the call stack (recursion).